client/landing/stepper/declarative-flow/internals/components/boot/index.tsx

```diff
@@ -0,0 +1,38 @@
+import { useLocale } from '@automattic/i18n-utils';
+import {
+	ComponentProps,
+	Suspense,
+	useEffect,
+	useState,
+	useTransition,
+	type FC,
+	type PropsWithChildren,
+} from 'react';
+import { useFlowLocale } from 'calypso/landing/stepper/hooks/use-flow-locale';
+
+interface Props extends PropsWithChildren {
+	fallback: ComponentProps< typeof Suspense >[ 'fallback' ];
+}
+
+//* This component is used to ensure that all required data is loaded before rendering the children.
+export const Boot: FC< Props > = ( { children, fallback } ) => {
+	const [ isReady, setIsReady ] = useState( false );
+	const [ isPending, setTransition ] = useTransition();
+
+	const locale = useLocale();
+	const newLocale = useFlowLocale();
```

Comment:
I am concerned that these are not the right values to look at, largely because it feels to me like `useFlowLocale()` is effectively a heuristic to work out what the locale "should" be. Might it not be better to get the `ui.localeSlug` value from the Calypso state as a primary indicator of the locale we expect to be loaded?

Reply:
@daledupreez 
Are you saying the  'ui.language.localeSlug' ? 

if yes.

This PR depended on #91935, where I [already updated ](https://github.com/Automattic/wp-calypso/pull/91935/files#diff-85d3d7a4b37f12ce63cd1f78a805154f3ed7db157b4bddcf2ff73796470614e2R5)`useFlowLocale` to really only on `'ui.language.localeSlug' to identify what locale "should" be. 

Is it answering your suggestion?

Reply:
Excellent! I missed that change in the bigger PR, and that this PR was based on the branch for that PR, too.

My concern is no longer relevant. 🥳

Reply:
As a related comment, it might be worth naming the two variables in a way that describes them a bit more. Maybe using `loadedLocale` and `selectedLocale` along with a comment?
```suggestion
	// There is a delay between switching the locale and the translations being fully loaded, so we're checking they match.
	const loadedLocale = useLocale();
	const selectedLocale = useFlowLocale();
```